### PR TITLE
Fix for deprecation warning std::not2

### PR DESCRIPTION
**Pull request by dparanic**

### CODE REVIEW

File: include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h [hunk 1]
```diff
@@ -65,7 +65,11 @@ exclusive_scan_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIter
     flags[0] = 1;
 
     transform(::std::forward<Policy>(policy), first1, last1 - 1, first1 + 1, _flags.get() + 1,
-              ::std::not2(binary_pred));
+              [binary_pred](
```
Comment (MikeDvorskiy) [inline include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h:68]: Actually you don't need to introduce new lambda-wrapper here.
There is `class __not_pred` defined in the file oneDPL\include\oneapi\dpl\pstl\utils.h

**Review by MikeDvorskiy — CHANGES_REQUESTED** [review]
1. the `__not_pred` class should be used instead (everywhere)
2. I guess the title should be corrected - I cant find any natch of `std::not1`

File: include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h (hunk 1 above)
Comment (dparanic) [reply include/oneapi/dpl/internal/exclusive_scan_by_segment_impl.h]: Applied.

**Review by MikeDvorskiy — DISMISSED** [review]

**Comment by dsobolev-dev** [comment]
We inject some deprecated/removed functionality as a part of [Tested Standard C++ APIs](https://docs.oneapi.io/versions/latest/onedpl/tested_standard_cpp_api.html), e.g. these: https://godbolt.org/z/a41qd6xaT. I suggest turning them off for c++20 and newer. It is not clear what to do with c++17: there may be warnings, but these APIs can be used on a user's side. Let's create an issue for the later discussion.

**Review by MikeDvorskiy — APPROVED** [review]

**Review by dsobolev-dev — APPROVED** [review]

**Pull request merged**